cmd/openshift-install/main.go

```diff
@@ -102,4 +107,35 @@ func runRootCmd(cmd *cobra.Command, args []string) {
 	if err != nil {
 		logrus.Fatal(errors.Wrap(err, "invalid log-level"))
 	}
+
+	if version.CommitDate != "" {
+		logOldInstallerNotification(version.CommitDate)
+	}
+}
+
+func logOldInstallerNotification(commitDate string) {
+	commit, err := time.Parse("2006-01-02", commitDate)
```

Comment:
Secondly, I don't really know how art sets the commit id and other details probably through some env variables.

I think that if the installer isn't changing for months even if release image is changing because some other component is getting updated is a pretty common case for our releases.

So this commit time based info is not correct.

Reply:
> I think that if the installer isn't changing for months even if release image is changing because some other component is getting updated is a pretty common case for our releases.

Once we hit GA, we seem to be rebuilding the installer with each release, even if there were no code changes on the installer side:

```console
$ for i in $(seq 0 16); do BUILD="$(oc image info -o json "$(oc adm release info --image-for=installer "quay.io/openshift-release-dev/ocp-release:4.4.${i}-x86_64")" | jq -r '.config.config.Labels | .["build-date"] + " " + .["io.openshift.build.commit.id"]')"; echo "4.4.$i ${BUILD}"; done
4.4.0 2020-04-26T20:06:07.506202 78b817ceb7657f81176bbe182cc6efc73004c841
4.4.1 2020-04-26T20:06:07.506202 78b817ceb7657f81176bbe182cc6efc73004c841
4.4.2 2020-04-26T20:06:07.506202 78b817ceb7657f81176bbe182cc6efc73004c841
4.4.3 2020-04-26T20:06:07.506202 78b817ceb7657f81176bbe182cc6efc73004c841
4.4.4 2020-05-12T06:36:05.489477 0c7f28d8642b8e9bf330038423120c5cf6a4bf91
4.4.5 2020-05-18T09:05:47.343422 15eac3785998a5bc250c9f72101a4a9cb767e494
4.4.6 2020-05-25T22:40:28.757122 99e1dc030910ccd241e5d2563f27725c0d3117b0
4.4.7 2020-06-01T19:39:57.237729 37d024ff54e3d2caf853ced640453222546be935
4.4.8 2020-06-08T06:44:29.165762 1541bf917973186bbab6a5f895f08db4334a5d9a
4.4.9 2020-06-16T03:24:46.264045 1541bf917973186bbab6a5f895f08db4334a5d9a
4.4.10 2020-06-21T18:10:09.339544 927b6d73cfa945abd6b48877db832fc76166a5c5
4.4.11 2020-06-29T05:07:03.141839 db69e0456f2f7d6b937a8e88fc1ee6be32bf61fd
4.4.12 2020-07-06T05:54:29.436475 db69e0456f2f7d6b937a8e88fc1ee6be32bf61fd
4.4.13 2020-07-12T03:41:16.238906 db69e0456f2f7d6b937a8e88fc1ee6be32bf61fd
4.4.14 2020-07-17T20:20:47.481039 88d06658198ec1eb3c8c8dc56d851b3ff4e3021f
4.4.15 2020-07-23T22:09:06.050116 88d06658198ec1eb3c8c8dc56d851b3ff4e3021f
4.4.16 2020-07-29T15:27:05.248605 88d06658198ec1eb3c8c8dc56d851b3ff4e3021f
$ curl -sH 'Accept:application/json' 'https://api.openshift.com/api/upgrades_info/v1/graph?channel=fast-4.4' | jq -r '.nodes[].version' | sort -V | grep '^4\.4\.' | head -n1
4.4.3
```

So 4.4 never recycled an installer build since it went GA in 4.4.3.  I guess it's possible that the "installer didn't change" releases always overlapped with underlying base-image changes.  Or maybe we always have underlying base-image changes, even when the installer changes too.  So while it would be good to double-check with ART before landing this, and yeah, looks like we should be using build-time instead of commit-time, but I think the time-based approach still has legs.